pkg/controller/component/synthesize_component.go

```diff
@@ -327,6 +332,44 @@ func buildLabelsAndAnnotations(compDef *appsv1alpha1.ComponentDefinition, comp *
 	}
 }
 
+func mergeUserDefinedEnv(synthesizedComp *SynthesizedComponent, comp *appsv1alpha1.Component) error {
+	if comp == nil || len(comp.Spec.Env) == 0 {
+		return nil
+	}
+
+	vars := map[string]bool{}
+	for _, v := range comp.Spec.Env {
+		if vars[v.Name] {
+			return fmt.Errorf("duplicated user-defined env var %s", v.Name)
+		}
+		vars[v.Name] = true
+	}
+
+	check := func(containers []corev1.Container) error {
+		for _, c := range containers {
+			for _, v := range c.Env {
+				if vars[v.Name] {
+					return fmt.Errorf("duplicated env var %s", v.Name)
```

Comment:
If comp.Spec.Env does not allow overriding containers[*].Env, will the feature requested in https://github.com/apecloud/kubeblocks/discussions/7502 be unachievable?

Reply:
fixed